### PR TITLE
Inhibit clipping when dropdown is opened

### LINKED ISSUE #9379: Dropdown width is truncated.

### PR DESCRIPTION
**Pull request by kazcw**
### Pull Request Description

Fixes #9379.

### Important Notes

- The existence-registry could be implemented with a counter, but a set is more debuggable and the performance cost is negligible.

### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [ ] The documentation has been updated, if necessary.
- [ ] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [ ] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [ ] Unit tests have been written where possible.
  - [ ] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: app/gui2/src/components/widgets/DropdownWidget.vue [hunk 1]
```diff
@@ -12,6 +13,14 @@ enum SortDirection {
...
 const emit = defineEmits<{ click: [index: number, keepOpen: boolean] }>()
 
+const tree = injectWidgetTree()
+
+let endClippingInhibition: (() => void) | undefined
+onMounted(() => {
+  endClippingInhibition = tree.inhibitClipping()
+})
+onUnmounted(() => endClippingInhibition?.())
+
```
Comment (farmaazon) [inline app/gui2/src/components/widgets/DropdownWidget.vue]: This should be part of WidgetSelection IMO. DropDownWidget itself may be a general-usage widget, and be used outside GraphNode (so there will be no injectWidgetTree)

File: app/gui2/src/providers/widgetTree.ts [hunk 2]
```diff
@@ -5,6 +5,20 @@ import { Ast } from '@/util/ast'
 import type { Icon } from '@/util/iconName'
 import { computed, proxyRefs, type Ref } from 'vue'
 
+function makeExistenceRegistry(onChange: (anyExist: boolean) => void) {
```
Comment (farmaazon) [inline app/gui2/src/providers/widgetTree.ts:8]: The function returning lambda returning lambda without any names is hard to understands.

Let's instead return object with "register" function with documentation, that "register" returns "unregister" handle... or let it be again an object with "unregister" field.

File: app/gui2/src/components/GraphEditor/widgets/WidgetApplication.vue [hunk 3]
```diff
@@ -48,7 +48,10 @@ export const widgetDefinition = defineWidget(ArgumentApplicationKey, {
       <NodeWidget :input="WidgetInput.FromAst(application.infixOperator)" />
     </div>
     <Transition name="collapse-argument">
-      <div v-if="tree.extended || !application.argument.hideByDefault" class="argument">
+      <div
+        v-if="tree.extended || !application.argument.hideByDefault"
+        class="argument overridableClipState"
+      >
```
Comment (farmaazon) [inline app/gui2/src/components/GraphEditor/widgets/WidgetApplication.vue:54]: Is it the only place which may clip drop-down? Could node clip the leftmost drop-down?

**Review by farmaazon — APPROVED** [review]

**Comment by farmaazon** [comment]
And one more thing: please add documentation explaining why do we have those clipping inhibitors.

File: app/gui2/src/components/GraphEditor/widgets/WidgetApplication.vue (hunk 3 above)
Comment (kazcw) [reply app/gui2/src/components/GraphEditor/widgets/WidgetApplication.vue:54]: This prevents the clipping required by the hidden-placeholders animation from interfering with the dropdown. I don't think we have any other animations that could clip the dropdown, but if we do this solution could be applied to them. If any widget is enabling clipping on its children statically, that would be a separate problem with a different solution (the widget must not do that).

**Pull request merged**